vnext/Microsoft.ReactNative/ReactRootView.cpp

```diff
@@ -481,4 +560,33 @@ void ReactRootView::RemoveChildAt(uint32_t index) {
   Children().RemoveAt(RNIndexToXamlIndex(index));
 }
 
+bool IsCtrlShiftI(winrt::Windows::System::VirtualKey key) noexcept {
+  return (
+      key == winrt::Windows::System::VirtualKey::I &&
+      ::Microsoft::ReactNative::IsModifiedKeyPressed(
+          winrt::CoreWindow::GetForCurrentThread(), winrt::Windows::System::VirtualKey::Shift) &&
+      ::Microsoft::ReactNative::IsModifiedKeyPressed(
+          winrt::CoreWindow::GetForCurrentThread(), winrt::Windows::System::VirtualKey::Control));
+}
+
+void ReactRootView::SetupDevToolsShortcut() noexcept {
+  if (auto xamlRoot = XamlRoot()) {
+    if (std::find(m_subscribedDebuggerRoots.begin(), m_subscribedDebuggerRoots.end(), xamlRoot) ==
+        m_subscribedDebuggerRoots.end()) {
+      if (auto rootContent = xamlRoot.Content()) {
+        m_subscribedDebuggerRoots.push_back(xamlRoot);
+        rootContent.KeyDown(
+            [weakThis = this->get_weak()](const auto & /*sender*/, const xaml::Input::KeyRoutedEventArgs &args) {
+              if (const auto strongThis = weakThis.get()) {
+                if (IsCtrlShiftI(args.Key())) {
+                  ::Microsoft::ReactNative::GetSharedDevManager()->OpenDevTools(
+                      winrt::to_string(strongThis->m_reactNativeHost.InstanceSettings().BundleAppId()));
```

Comment:
This is a hack, we probably need to put this somewhere else, but I couldn't think of a good place that both had access to DevSettings and all active XamlRoots, and I was being a bit lazy not wanting to wire this through the ReactNativeHost.